src/Sparrow/Logging/LoggingSource.cs

```diff
@@ -239,12 +284,61 @@ private FileStream GetNewStream(long maxFileSize)
                                nextLogNumber.ToString("000", CultureInfo.InvariantCulture) + ".log";
                 if (File.Exists(fileName) && new FileInfo(fileName).Length >= maxFileSize)
                     continue;
+
+                LimitLogSize(logFiles, logGzFiles);
                 var fileStream = SafeFileStream.Create(fileName, FileMode.Append, FileAccess.Write, FileShare.Read, 32 * 1024, false);
                 fileStream.Write(_headerRow, 0, _headerRow.Length);
                 return fileStream;
             }
         }
 
+        private void LimitLogSize(string[] logFiles, string[] logGzFiles)
+        {
+            var logFilesInfo = logFiles.Select(f => new FileInfo(f));
+            var logGzFilesInfo = logGzFiles.Select(f => new FileInfo(f));
+            var totalLogSize = logFilesInfo.Sum(i => i.Length) + logGzFilesInfo.Sum(i => i.Length);
+
+            foreach (var log in logGzFilesInfo.Reverse())
+            {
+                if (totalLogSize > RetentionSize)
```

Comment:
This should be before both of the loops to avoid checking that each time

Reply:
Actually, I forgot to decrement the `totalLogSize` value every deletion...

Reply:
Can we have a test for this? We can create new instance of LoggingSource inside a test and check if everything is working as expected.